crates/solver/src/solver/balancer_sor_solver.rs

```diff
@@ -76,8 +76,12 @@ impl SingleOrderSolving for BalancerSorSolver {
             }
         };
 
-        if !quote_respects_limit_price(&quote, &order) {
-            tracing::debug!("Order limit price not respected");
+        let (executed_sell, executed_buy) = match order.kind {
```

Comment:
nit: We already compute:
```rust
        let (quoted_sell_amount, quoted_buy_amount) = match order.kind {
            OrderKind::Sell => (quote.swap_amount, quote.return_amount),
            OrderKind::Buy => (quote.return_amount, quote.swap_amount),
        };
```

A couple lines below, maybe we don't need to re-match on `order.kind` and only do this once?

Reply:
Somehow missed that. Fixed.